library/modules/Gui.cpp

```diff
@@ -130,529 +130,307 @@ static std::string getNameChunk(virtual_identity *id, int start, int end)
-        break;
-    }
-}
-
-DEFINE_GET_FOCUS_STRING_HANDLER(layer_overall_health)
-{
-    auto list1 = getLayerList(screen, 0);
-    if (!list1) return;
-
-    focus += "/Units";
+    return std::find_if(currentFocus.begin(), currentFocus.end(), [&focusString](std::string item) {
+        //Transform the each vector item to lower temporally 
+        return focusString == toLower(item);
+    }) != currentFocus.end();
 }
 
-DEFINE_GET_FOCUS_STRING_HANDLER(tradegoods)
+std::vector<std::string> Gui::getFocusString(df::viewscreen* top)
 {
-    if (!screen->has_traders || screen->is_unloading)
-        focus += "/NoTraders";
-    else if (screen->in_edit_count)
-        focus += "/EditCount";
-    else
-        focus += (screen->in_right_pane ? "/Items/Broker" : "/Items/Trader");
-}
-
-DEFINE_GET_FOCUS_STRING_HANDLER(layer_assigntrade)
-{
-    auto list1 = getLayerList(screen, 0);
-    auto list2 = getLayerList(screen, 1);
-    if (!list1 || !list2) return;
-
-    int list_idx = vector_get(screen->visible_lists, list1->cursor, (int16_t)-1);
-    unsigned num_lists = sizeof(screen->lists)/sizeof(screen->lists[0]);
-    if (unsigned(list_idx) >= num_lists)
-        return;
-
-    if (list1->active)
-        focus += "/Groups";
-    else
-        focus += "/Items";
-}
-
-DEFINE_GET_FOCUS_STRING_HANDLER(stores)
-{
-    if (!screen->in_right_list)
-        focus += "/Categories";
-    else if (screen->in_group_mode)
-        focus += "/Groups";
-    else
-        focus += "/Items";
-}
-
-DEFINE_GET_FOCUS_STRING_HANDLER(layer_stockpile)
-{
-    auto list1 = getLayerList(screen, 0);
-    auto list2 = getLayerList(screen, 1);
-    auto list3 = getLayerList(screen, 2);
-    if (!list1 || !list2 || !list3 || !screen->settings) return;
+    std::vector<std::string> focusStrings;
 
-    auto group = screen->cur_group;
-    if (group != vector_get(screen->group_ids, list1->cursor))
-        return;
-
-    focus += "/" + enum_item_key(group);
-
-    auto bits = vector_get(screen->group_bits, list1->cursor);
-    if (bits.whole && !(bits.whole & screen->settings->flags.whole))
-    {
-        focus += "/Off";
-        return;
-    }
-
-    focus += "/On";
-
-    if (list2->active || list3->active || screen->list_ids.empty()) {
-        focus += "/" + enum_item_key(screen->cur_list);
-
-        if (list3->active)
-            focus += (screen->item_names.empty() ? "/None" : "/Item");
-    }
-}
-
-DEFINE_GET_FOCUS_STRING_HANDLER(locations)
-{
-    focus += "/" + enum_item_key(screen->menu);
-}
-
-DEFINE_GET_FOCUS_STRING_HANDLER(jobmanagement)
-{
-    focus += (screen->in_max_workshops ? "/MaxWorkshops" : "/Main");
-}
-
-DEFINE_GET_FOCUS_STRING_HANDLER(workquota_condition)
-{
-    focus += "/" + enum_item_key(screen->mode);
-    if (screen->item_count_edit)
-        focus += "/EditCount";
-}
-*/
-
-std::string Gui::getFocusString(df::viewscreen *top)
-{
     if (!top)
-        return "";
+        return focusStrings;
 
     if (dfhack_viewscreen::is_instance(top))
```

Comment:
```suggestion
    if (dfhack_viewscreen::is_instance(top))
        top = static_cast<dfhack_viewscreen*>(top)->getFocusedScreen();

    if (dfhack_viewscreen::is_instance(top))
```

Reply:
We'll need to check here whether we fall through ZScreens. getFocusedScreen() will need to be implemented for DFHack's viewscreen implementation in the Screen module